dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsExtendedApi.java

```diff
@@ -83,32 +83,58 @@ public Response toolsOrgGet(
+    @ApiResponses(value = { @ApiResponse(code = HttpStatus.SC_OK, message = WorkflowsIndexSearch.OK_RESPONSE, response = String.class) })
+    @Operation(operationId = WorkflowsIndexSearch.OPERATION_ID, summary = WorkflowsIndexSearch.SUMMARY, description = WorkflowsIndexSearch.DESCRIPTION, responses = {
+            @io.swagger.v3.oas.annotations.responses.ApiResponse(responseCode = HttpStatus.SC_OK + "", description = WorkflowsIndexSearch.OK_RESPONSE, content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(implementation = String.class)))
+    })
+    public Response workflowsIndexSearch(@ApiParam(value = "elastic search query", required = true) String query,
+                                     @Context UriInfo uriInfo, @Context SecurityContext securityContext) {
+        return delegate.workflowsIndexSearch(query, uriInfo != null ? uriInfo.getQueryParameters() : null, securityContext);
+    }
+
+    @POST
+    @Path("/entries/index")
     @UnitOfWork
     @RolesAllowed({"curator", "admin"})
     @Produces({ MediaType.TEXT_PLAIN })
-    @ApiOperation(value = ToolsIndexGet.SUMMARY, notes = ToolsIndexGet.DESCRIPTION, authorizations = {
+    @ApiOperation(value = EntriesIndexGet.SUMMARY, notes = EntriesIndexGet.DESCRIPTION, authorizations = {
         @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = Integer.class)
-    @ApiResponses(value = { @ApiResponse(code = HttpStatus.SC_OK, message = ToolsIndexGet.OK_RESPONSE) })
-    @Operation(operationId = ToolsIndexGet.SUMMARY, summary = ToolsIndexGet.SUMMARY, description = ToolsIndexGet.DESCRIPTION, security = @SecurityRequirement(name = ResourceConstants.OPENAPI_JWT_SECURITY_DEFINITION_NAME), responses = {
-        @io.swagger.v3.oas.annotations.responses.ApiResponse(responseCode = HttpStatus.SC_OK + "", description = ToolsIndexGet.OK_RESPONSE, content = @Content(mediaType = MediaType.TEXT_PLAIN, schema = @Schema(implementation = Integer.class)))
+    @ApiResponses(value = { @ApiResponse(code = HttpStatus.SC_OK, message = EntriesIndexGet.OK_RESPONSE) })
+    @Operation(operationId = EntriesIndexGet.SUMMARY, summary = EntriesIndexGet.SUMMARY, description = EntriesIndexGet.DESCRIPTION, security = @SecurityRequirement(name = ResourceConstants.OPENAPI_JWT_SECURITY_DEFINITION_NAME), responses = {
+        @io.swagger.v3.oas.annotations.responses.ApiResponse(responseCode = HttpStatus.SC_OK + "", description = EntriesIndexGet.OK_RESPONSE, content = @Content(mediaType = MediaType.TEXT_PLAIN, schema = @Schema(implementation = Integer.class)))
     })
-    public Response toolsIndexGet(@ApiParam(hidden = true) @Parameter(hidden = true) @Auth User user, @Context SecurityContext securityContext)
+    public Response entriesIndexGet(@ApiParam(hidden = true) @Parameter(hidden = true) @Auth User user, @Context SecurityContext securityContext)
```

Comment:
The main TRS repo has the concept of a toolClass https://github.com/ga4gh/tool-registry-service-schemas/blob/develop/openapi/ga4gh-tool-discovery.yaml#L147 
This is a parameter which filters down tools by subclass (DockstoreTool, Workflow).
Would it be possible to have the same parameter here for consistency (leaving it blank or null to get entries/all)

The Strings for the ToolClasses are available here https://github.com/dockstore/dockstore/blob/b64cb4e6655e5945582eb3192ab3bbf0c3e3092f/dockstore-webservice/src/main/java/io/openapi/api/impl/ToolClassesApiServiceImpl.java#L34